azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java

```diff
@@ -28,7 +40,34 @@
      * @throws AzureLoginFailureException when there are some errors during login.
      */
     public static AzureCredential oAuthLogin(AzureEnvironment env) throws AzureLoginFailureException {
-        throw new UnsupportedOperationException("Not implemented");
+        try {
+            final WebAppServer webAppServer = new WebAppServer();
+            final String redirectUrl = webAppServer.getUrl();
+            final URI redirectUri = new URI(redirectUrl);
+            try {
+                final String authorizationUrl = authorizationUrl(env, redirectUrl);
+                try {
+                    Desktop.getDesktop().browse(new URL(authorizationUrl).toURI());
```

Comment:
Before calling the browse(), please check whether browsing is supported using `Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)`

Reply:
good advice, will use it to test browser